### PR TITLE
mysql: update to 5.7.12

### PR DESCRIPTION
**Pull request by lrusak**
mysql 5.7.12 needs the exact boost version 1_59_0. So downgrading boost is the simplest solution.

mysql 5.7 bring tls 1.2 encryption.

@MilhouseVH a clean build with this would be appreciated.

### CODE REVIEW

**Comment by lrusak** [comment]
Updated, thanks for testing @MilhouseVH 👍

**Comment by vpeter4** [comment]
It is possible to instruct mysql 5.7.12 to download correct boost version when building.

**Comment by lrusak** [comment]
@vpeter4 yes it is, however this is a lot of extra bandwidth when you have to build all the projects :)

**Comment by vpeter4** [comment]
Agree. If boost on a system will need to upgrade back then for mysql should be used different solution.
Because I was rebuilding this few times I download the package once and unpack the content to required folder.

**Comment by stefansaraev** [comment]
what if something else needs "exact" boost 1.58.0 tomorrow?

doesnt this work ? : -DDOWNLOAD_BOOST=1 -DWITH_BOOST=$ROOT/$PKG_BUILD/boost_build

**Comment by vpeter4** [comment]
Yes, downloading boost works from mysql itself.

**Comment by stefansaraev** [comment]
so, mysql now requires boost for a feature you wont ever use (GIS). why not use 5.5 or 5.6 instead ?

**Comment by lrusak** [comment]
because tls 1.2
http://dev.mysql.com/doc/relnotes/mysql/5.7/en/news-5-7-10.html

**Comment by stefansaraev** [comment]
what's the benefit of using tls in your local network ?

**Comment by lrusak** [comment]
maybe they want some shared cloud mysql database. who knows 😄

**Comment by stefansaraev** [comment]
so TLSv1 that is available in 5.6 is not enough?

**Comment by MilhouseVH** [comment]
The `boost` dependency is unfortunate but is this likely to be a problem when the trend is to move away from `boost`? A quick scan shows only `mpd` with a `boost` dependency, and `mpd` should be able to use 1_59_0.

**Comment by sopparus** [comment]
since boost is only used for GIS it would be safe to make mysql ignore the 1.59 check and compile with  1.60. I have tried that and it works. (just an option)

**Comment by lrusak** [comment]
thanks @sopparus . I have dropped the boost commit and updated mysql to work with boost 1_60_0

File: packages/databases/mysql/package.mk [hunk 1]
```diff
@@ -1,114 +1,117 @@
...
-TARGET_CFLAGS="$TARGET_CFLAGS -fPIC -DPIC"
+post_unpack() {
+  sed -i 's|OPENSSL_MAJOR_VERSION STREQUAL "1"|OPENSSL_MAJOR_VERSION STREQUAL "2"|' $ROOT/$PKG_BUILD/cmake/ssl.cmake
+  sed -i 's|GET_TARGET_PROPERTY(LIBMYSQL_OS_OUTPUT_NAME libmysql OUTPUT_NAME)|SET(LIBMYSQL_OS_OUTPUT_NAME "mysqlclient")|' $ROOT/$PKG_BUILD/scripts/CMakeLists.txt
+  sed -i "s|COMMAND comp_err|COMMAND $ROOT/$TOOLCHAIN/bin/comp_err|" $ROOT/$PKG_BUILD/extra/CMakeLists.txt
+  sed -i "s|COMMAND comp_sql|COMMAND $ROOT/$TOOLCHAIN/bin/comp_sql|" $ROOT/$PKG_BUILD/scripts/CMakeLists.txt
+  sed -i "s|COMMAND gen_lex_hash|COMMAND $ROOT/$TOOLCHAIN/bin/gen_lex_hash|" $ROOT/$PKG_BUILD/sql/CMakeLists.txt
 
-PKG_CONFIGURE_OPTS_HOST="--with-zlib-dir=$ROOT/$TOOLCHAIN"
+  sed -i "s|BOOST_MINOR_VERSION EQUAL 59|BOOST_MINOR_VERSION EQUAL 60|" $ROOT/$PKG_BUILD/cmake/boost.cmake
```
Comment (stefansaraev) [inline packages/databases/mysql/package.mk]:
```
sed -i '/^IF(NOT BOOST_MINOR_VERSION.*$/,/^ENDIF()$/d' $ROOT/$PKG_BUILD/cmake/boost.cmake
```
even better :)

**Comment by lrusak** [comment]
@stefansaraev thanks 👍

**Comment by MindTooth** [comment]
MariaDB?

**Comment by MilhouseVH** [comment]
> MariaDB?

Maybe if/when Kodi officially recognises and supports MariaDB, but for now it's MySQL only.

If you want to use MariaDB instead of MySQL with Kodi, you're on your own, and there have been times when MariaDB hasn't worked with Kodi (MariaDB bugs etc.), so it's additional risk/pain.

**Comment by MindTooth** [comment]
I see @MilhouseVH :-)  Thank you for taking the time explaining it.

**Comment by lrusak** [comment]
@MilhouseVH any negative effects?

**Comment by MilhouseVH** [comment]
Nope, all good.

**Pull request merged**